src/core/pointcloud/qgsremotecopcpointcloudindex.cpp

```diff
@@ -0,0 +1,373 @@
+  attributes.push_back( QgsPointCloudAttribute( "ScanDirectionFlag", QgsPointCloudAttribute::Char ) );
+  attributes.push_back( QgsPointCloudAttribute( "EdgeOfFlightLine", QgsPointCloudAttribute::Char ) );
+  attributes.push_back( QgsPointCloudAttribute( "Classification", QgsPointCloudAttribute::Char ) );
+  attributes.push_back( QgsPointCloudAttribute( "ScanAngleRank", QgsPointCloudAttribute::Short ) );
+  attributes.push_back( QgsPointCloudAttribute( "UserData", QgsPointCloudAttribute::Char ) );
+  attributes.push_back( QgsPointCloudAttribute( "PointSourceId", QgsPointCloudAttribute::UShort ) );
+  attributes.push_back( QgsPointCloudAttribute( "GpsTime", QgsPointCloudAttribute::Double ) );
+
+  switch ( mCopcHeader.pointFormat() )
+  {
+    case 6:
+      break;
+    case 7:
+      attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Red" ), QgsPointCloudAttribute::UShort ) );
+      attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Green" ), QgsPointCloudAttribute::UShort ) );
+      attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Blue" ), QgsPointCloudAttribute::UShort ) );
+      break;
+    case 8:
+      attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Red" ), QgsPointCloudAttribute::UShort ) );
+      attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Green" ), QgsPointCloudAttribute::UShort ) );
+      attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Blue" ), QgsPointCloudAttribute::UShort ) );
+      attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "NIR" ), QgsPointCloudAttribute::UShort ) );
+      break;
+    default:
+      return false;
+  }
+
+  QVector<QgsLazDecoder::ExtraBytesAttributeDetails> extrabyteAttributes = QgsLazDecoder::readExtraByteAttributesFromVlr( mExtraBytesVlr, mCopcHeader.point_record_length );
+  for ( QgsLazDecoder::ExtraBytesAttributeDetails attr : extrabyteAttributes )
+  {
+    attributes.push_back( QgsPointCloudAttribute( attr.attribute, attr.type ) );
+  }
+
+  setAttributes( attributes );
+
+  const double xmin = mCopcInfoVlr.center_x - mCopcInfoVlr.halfsize;
+  const double ymin = mCopcInfoVlr.center_y - mCopcInfoVlr.halfsize;
+  const double zmin = mCopcInfoVlr.center_z - mCopcInfoVlr.halfsize;
+  const double xmax = mCopcInfoVlr.center_x + mCopcInfoVlr.halfsize;
+  const double ymax = mCopcInfoVlr.center_y + mCopcInfoVlr.halfsize;
+  const double zmax = mCopcInfoVlr.center_z + mCopcInfoVlr.halfsize;
+
+  mRootBounds = QgsPointCloudDataBounds(
+                  ( xmin - mOffset.x() ) / mScale.x(),
+                  ( ymin - mOffset.y() ) / mScale.y(),
+                  ( zmin - mOffset.z() ) / mScale.z(),
+                  ( xmax - mOffset.x() ) / mScale.x(),
+                  ( ymax - mOffset.y() ) / mScale.y(),
+                  ( zmax - mOffset.z() ) / mScale.z()
+                );
+
+  double calculatedSpan = nodeMapExtent( root() ).width() / mCopcInfoVlr.spacing;
+  mSpan = calculatedSpan;
+
```

Comment:
A lot of the code here seems duplicated from the local COPC implementation. It may be useful to either move it to some utils class - or - have a copc point cloud index class that would contain members and functions that are common for both local and remote COPC index, with a class hierarchy like this:
```
   QgsPointCloudIndex
                 ^
                 |
QgsCopcPointCloudIndex
                 ^
                 |
   ---------------------------------
  |                                |
QgsLocalCopcPointCloudIndex   QgsRemoteCopcPointCloudIndex